### PR TITLE
Only load the configured switches for services

### LINKED ISSUE #1308: Only pre-load the configured switches
To reduce memory usage only preload the configured switches instead of all.

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

To avoid using too much memory only load switches that are configured
## Impacts

All pf services that preload switches
## NEWS file entries

Enhancements
++++++++++++
- Only preload switch modules that are configured to reduce memory usage

Bug Fixes
+++++++++
Fixes #1308 
## Delete branch after merge

NO

NOTES
++++++++
Remove your old conf/httpd.conf/*.tt files as they the example file have changed

### CODE REVIEW

File: lib/pf/WebAPI/InitHandler.pm [hunk 1]
```diff
@@ -57,14 +57,25 @@ Close connections to avoid any sharing of sockets
...
     db_disconnect();
+    $class->preloadSwitches();
     pf::CHI->clear_memoized_cache_objects;
-    pf::SwitchFactory::preLoadModules();
     return Apache2::Const::OK;
 }
 
+=head2 preloadSwitches
+
+TODO: documention
```
Comment (julsemaan) [inline lib/pf/WebAPI/InitHandler.pm]: image

File: lib/pf/WebAPI/InitHandler/Admin.pm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+=cut
+
+use strict;
+use warnings;
+
+use pf::SwitchFactory();
+
+use base qw(pf::WebAPI::InitHandler);
+
+sub preloadSwitches {
```
Comment (julsemaan) [inline lib/pf/WebAPI/InitHandler/Admin.pm:28]: doc

File: lib/pfconfig/namespaces/resource/SwitchTypesConfigured.pm [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+=cut
+
+use strict;
+use warnings;
+
+use base 'pfconfig::namespaces::resource';
+
+use NetAddr::IP;
+
+sub init {
```
Comment (julsemaan) [inline lib/pfconfig/namespaces/resource/SwitchTypesConfigured.pm:30]: doc here and below

File: lib/pfconfig/namespaces/resource/SwitchTypesConfigured.pm [hunk 4]
```diff
@@ -0,0 +1,74 @@
+package pfconfig::namespaces::resource::SwitchTypesConfigured;
```
Comment (julsemaan) [inline lib/pfconfig/namespaces/resource/SwitchTypesConfigured.pm:1]: This has to be a child resource of config::Switch

File: lib/pfconfig/namespaces/resource/SwitchTypesConfigured.pm [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+pfconfig::namespaces::resource::SwitchTypesConfigured
+
+=cut
+
+=head1 DESCRIPTION
+
+pfconfig::namespaces::resource::SwitchTypesConfigured
+
+This module creates the configuration hash of all the switches ranges (ex : 192.168.1.0/24)
```
Comment (julsemaan) [inline lib/pfconfig/namespaces/resource/SwitchTypesConfigured.pm]: No it doesn't

Comment (julsemaan) [inline lib/pfconfig/namespaces/resource/SwitchTypesConfigured.pm]: image

**Comment by jrouzierinverse** [comment]
Everything has been taken care of please take another look

**Comment by jrouzierinverse** [comment]
Bump

**Comment by julsemaan** [comment]
I would hapily merge it but Github is b0rken. Merge it when you can

image

**Pull request merged**